data_preparation/data_cleaning.py

```diff
@@ -130,30 +128,24 @@ def clean_round1(df):
                         'combined': 'float', 'seller_rating': 'float'
                         })
 
-    df['make'] = df.make.map(lambda x: x.strip().lower().replace(' ', '-'), na_action='ignore').astype('category')
-    df['model'] = df.model.map(lambda x: x.strip().lower().replace(' ', '-'), na_action='ignore').astype('category')
-    df['trim'] = df.trim.map(lambda x: x.strip().lower().replace(' ', '-'), na_action='ignore')
-    df['body_type'] = df.body_type.map(lambda x: x.strip().lower().replace(' ', '_'), na_action='ignore').astype('category')
-    df['transmission'] = df.transmission.map(lambda x: x.strip().lower().replace(' ', '_'), na_action='ignore').astype('category')
-    df['fuel_type'] = df.fuel_type.map(lambda x: x.strip().lower().replace(' ', '_'), na_action='ignore').astype('category')
-    df['insurance_group'] = df.insurance_group.map(lambda x: x.strip().lower().replace(' ', '_'), na_action='ignore').astype('category')
-    df['seller_segment'] = df.seller_segment.map(lambda x: x.strip().lower().replace(' ', '_'), na_action='ignore').astype('category')
-    df['region'] = df.region.map(lambda x: x.strip().lower().replace(' ', '_'), na_action='ignore').astype('category')
-    df['county'] = df.county.map(lambda x: x.strip().lower().replace(' ', '_'), na_action='ignore').astype('category')
-    df['town'] = df.town.map(lambda x: x.strip().lower().replace(' ', '_'), na_action='ignore').astype('category')
+    cols = ['make','model','trim','body_type','transmission','fuel_type','insurance_group','seller_segment','region','county','town']
+    sep = defaultdict(lambda:'_', {'make':'-', 'model':'-', 'trim':'-'})
```

Comment:
You know how i used _ and - i don't think there was an actual reason other than preferring how it looked in make, model and trim columns. ie mercedes-benz vs mercedes_benz. I think you could change it all to _ if that makes the code simpler

Reply:
Hmm okay, I'll look at it again, I thought I remembered there being a place where you relied on the dash as opposed to the underscore, but could be wrong. Also you don't actually use this function in the preprocessing script anyway?